x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -87,77 +124,263 @@ export function MachineLearningDataVisualizerTableProvider({ getService }: FtrPr
       const docCount = await testSubjects.getVisibleText(docCountFormattedSelector);
       expect(docCount).to.eql(
         docCountFormatted,
-        `Expected total document count to be '${docCountFormatted}' (got '${docCount}')`
+        `Expected field document count to be '${docCountFormatted}' (got '${docCount}')`
       );
     }
 
-    public async assertNumberRowContents(fieldName: string, docCountFormatted: string) {
-      const fieldType = ML_JOB_FIELD_TYPES.NUMBER;
-      await this.assertRowExists(fieldName);
-      await this.assertFieldDocCount(fieldName, docCountFormatted);
+    public async assertFieldDistinctValuesExist(fieldName: string) {
+      const selector = this.rowSelector(fieldName, 'mlDataVisualizerTableColumnDistinctValues');
+      await testSubjects.existOrFail(selector);
+    }
 
-      await this.expandRowDetails(fieldName, fieldType);
+    public async assertFieldDistributionExist(fieldName: string) {
+      const selector = this.rowSelector(fieldName, 'mlDataVisualizerTableColumnDistribution');
+      await testSubjects.existOrFail(selector);
+    }
 
-      await testSubjects.existOrFail(
-        `mlDataVisualizerFieldExpandedRow ${fieldName} ${fieldType} > mlNumberSummaryTable`
+    public async assertSearchPanelExist() {
+      await testSubjects.existOrFail(`mlDataVisualizerSearchPanel`);
+    }
+
+    public async assertFieldNameInputExists() {
+      await testSubjects.existOrFail('mlDataVisualizerFieldNameSelect');
+    }
+
+    public async assertFieldTypeInputExists() {
+      await testSubjects.existOrFail('mlDataVisualizerFieldTypeSelect');
+    }
+
+    public async assertSampleSizeInputExists() {
+      await testSubjects.existOrFail('mlDataVisualizerShardSizeSelect');
+    }
+
+    public async setSampleSizeInputValue(
+      sampleSize: number,
+      fieldName: string,
+      docCountFormatted: string
+    ) {
+      await this.assertSampleSizeInputExists();
+      await testSubjects.clickWhenNotDisabled('mlDataVisualizerShardSizeSelect');
+      await testSubjects.existOrFail(`mlDataVisualizerShardSizeOption ${sampleSize}`);
+      await testSubjects.click(`mlDataVisualizerShardSizeOption ${sampleSize}`);
+
+      await retry.tryForTime(5000, async () => {
+        await this.assertFieldDocCount(fieldName, docCountFormatted);
+      });
+    }
+
+    public async setMultiSelectFilter(
```

Comment:
This method and `removeMultiSelectFilter` could probably be moved to the common UI service for future re-use. But this can be done in a future PR.

Reply:
Moved these two methods to mlCommonUI here [`5581931` (#86071)](https://github.com/elastic/kibana/pull/86071/commits/55819316abe5e29c01932dbc71b53f51ece2f9d6)